### DISCUSSION TITLE
CFs restart on "roslaunch crazyswarm hover_swarm.launch" with new CF Firmware

### DISCUSSION THREAD

**Discussion by stikkx** [USC-ACTLab/crazyswarm#313] [Q&A]
I want to fly 9 CFs at the same time with LH2.0. Some of them rebooting after i start the Crazyswarmserver. It does not matter if the LH2.0 is on or off. Here is the Log from the Server:

`[ INFO] [1597211847.988366944]: Adding CF: cf11 (radio://0/100/2M/E7E7E7E70b, cf11)...
[ INFO] [1597211848.019028939]: CF ctor: 0.030588 s
[ INFO] [1597211848.023177527]: [cf11] SYS: ----------------------------
[ INFO] [1597211848.027086495]: [cf11] SYS: Crazyflie 2.1 is up and running!
[ INFO] [1597211848.030110747]: [cf11] SYS: Build 37:9e857db5011e (2020.06 +37) MODIFIED
[ WARN] [1597211848.031681076]: [cf11] Link Quality low (0.470000)
[ INFO] [1597211848.034050694]: [cf11] SYS: I am 0x20383347344D500700290028 and I have 1024KB of flash!
[ INFO] [1597211848.035708137]: [cf11] CFGBLK: v1, verification [OK]
[ INFO] [1597211848.037168994]: [cf11] DECK_CORE: 2 deck(s) found
[ INFO] [1597211848.039975651]: [cf11] DECK_CORE: Calling INIT on driver bcLedRing for deck 0
[ INFO] [1597211848.042923543]: [cf11] DECK_CORE: Calling INIT on driver bcLighthouse4 for deck 1
[ INFO] [1597211848.045329315]: [cf11] IMU: BMI088 Gyro I2C connection [OK].
[ INFO] [1597211848.047772594]: [cf11] IMU: BMI088 Accel I2C connection [OK]
[ INFO] [1597211848.050287615]: [cf11] IMU: BMP388 I2C connection [OK]
[ INFO] [1597211848.052866280]: [cf11] ESTIMATOR: Using Kalman (2) estimator
[ INFO] [1597211848.055466703]: [cf11] CONTROLLER: Using PID (1) controller
[ INFO] [1597211848.057841408]: [cf11] MTR-DRV: Using brushed motor driver
[ INFO] [1597211848.059432808]: [cf11] EEPROM: I2C connection [OK].
[ INFO] [1597211848.061730157]: [cf11] IMU: BMI088 gyro self-test [OK]
[ INFO] [1597211848.064125867]: [cf11] STAB: Wait for sensor calibration...
[ INFO] [1597211848.067007525]: [cf11] SYS: FreeLHFL: Lighthouse bootloader version: 2
[ INFO] [1597211848.068239943]: [cf11]  heap: 11760 bytes
[ INFO] [1597211848.069487505]: [cf11] STAB: Ready to fly.
[ INFO] [1597211848.072177716]: [cf11] LHFL: Firmware version 5 verified, booting deck!
[ INFO] [1597211848.073064828]: [cf11] Requesting parameters...
[ INFO] [1597211848.148415654]: Found variables in cache.
[ INFO] [1597211848.476891423]: [cf11] reqParamTOC: 0.457790 s (199 params)
[ INFO] [1597211848.476936965]: Requesting memories...
[ INFO] [1597211848.517343611]: Memories: 11
[ WARN] [1597211848.605124244]: [cf11] Link Quality low (0.640000)
[ INFO] [1597211848.641805807]: [cf11] Ready. Elapsed: 0.622694 s
[ INFO] [1597211848.641897864]: CF run: 0.622877 s
update commander/enHighLevel to 1
update kalman/resetEstimation to 1
update locSrv/extPosStdDev to 0.001
update locSrv/extQuatStdDev to 0.05
update ring/effect to 0
update ring/headlightEnable to 0
update ring/solidBlue to 0
update ring/solidGreen to 0
update ring/solidRed to 255
update stabilizer/controller to 2
update stabilizer/estimator to 2
update ctrlMel/i_range_m_z to 1500
update ctrlMel/i_range_xy to 2
update ctrlMel/i_range_z to 0.4
update ctrlMel/kR_xy to 70000
update ctrlMel/kR_z to 60000
update ctrlMel/kd_omega_rp to 200
update ctrlMel/kd_xy to 0.2
update ctrlMel/kd_z to 0.4
update ctrlMel/ki_m_z to 500
update ctrlMel/ki_xy to 0.05
update ctrlMel/ki_z to 0.05
update ctrlMel/kp_xy to 0.4
update ctrlMel/kp_z to 1.25
update ctrlMel/kw_xy to 20000
update ctrlMel/kw_z to 12000
update ctrlMel/mass to 0.032
update ctrlMel/massThrust to 132000
[ INFO] [1597211848.665440079]: [cf11] Update parameters
[ INFO] [1597211848.698459353]: [cf11] SYS: Assert failed .//src/modules/src/crtp.c:185
[ INFO] [1597211848.814064962]: [cf11] SYS: ----------------------------
[ INFO] [1597211848.816760588]: [cf11] SYS: Crazyflie 2.1 is up and running!
[ INFO] [1597211848.819668498]: [cf11] SYS: Build 37:9e857db5011e (2020.06 +37) MODIFIED
[ INFO] [1597211848.892489370]: [cf11] SYS: I am 0x20383347344D500700290028 and I have 1024KB of flash!
[ INFO] [1597211848.914540547]: [cf11] CFGBLK: v1, verification [OK]
[ WARN] [1597211848.938859728]: [cf11] Link Quality low (0.120000)
[ WARN] [1597211849.044141130]: [cf11] Link Quality low (0.240000)
[ INFO] [1597211849.080993689]: [cf11] DECK_CORE: 2 deck(s) found
[ INFO] [1597211849.084135539]: [cf11] DECK_CORE: Calling INIT on driver bcLedRing for deck 0
[ INFO] [1597211849.087492992]: [cf11] DECK_CORE: Calling INIT on driver bcLighthouse4 for deck 1
`

The CrazyRadio is about 3 Meter far away from the CFs and the LH2.0 are in the corners of the room and i measured 5 meter between them. The LH2.0 are mounted at an angle of 45°. I wrote the position to the firmware(newest firmware from bitcraze repo) and updated. Do you have a tip how i can debug or solve the problem ?

**Comment by whoenig** [comment]
I think you'll need to change the firmware a bit to avoid that problem See https://github.com/USC-ACTLab/crazyswarm/issues/182 and https://github.com/USC-ACTLab/crazyswarm/issues/240 for a discussion and details.
